### ISSUE TITLE
SortedSwitchCases Bug?

### ISSUE THREAD

**Issue by PompeiaPaetenari** [nicklockwood/SwiftFormat#829]
It looks like the style of sorting changed in the recent update, which I know fixed the tuple issue but now it looks weird (or so the PR comments point out).
Currently, we get this from 0.47.7:
```swift
  switch self {
    case let .nameAndDateOfBirth(user, _),
         let .name(user):
      return name
    case .dateOfBirth:
      return nil
    }
```
But are expecting to have, like in 0.47.5:
```swift
switch self {
  case let .name(user),
       let .nameAndDateOfBirth(user, _):
    return name
  case .dateOfBirth:
    return nil
  }
```

Basically, if the string is shorter, as seen with an r"\b", it would be earlier in the list, right?  That is my general understanding for sorting.  Or is this desired behaviour?

**Comment by nicklockwood** [comment]
It's actually now sorting by the concatenation of all identifiers in the case instead of just the first, but I guess it should ignore `user` and/or take delimiters into account. I'll try tweaking the algorithm a bit.

**Comment by nicklockwood** [comment]
@PompeiaPaetenari fixed in 0.47.9.
